src/components/SearchStock.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { setSearchTerm } from '../redux/stocks/stocksSlice';
+
+const SearchStock = () => {
+  const dispatch = useDispatch();
+  const handleChange = (e) => {
+    console.log('What is it? ', e.target.value);
+    dispatch(setSearchTerm(e.target.value));
+  };
+  return (
+    <div>
+      <input
+        type="text"
+        placeholder="Search stock Name..."
+        onChange={(e) => handleChange(e)}
+        className="search-box"
+      />
```

Comment:
👍 Your search functionality is working well! However, there's a minor usability improvement to consider:

- [ ] After a user has performed a search and then navigates back to the home page from a company details page currenty you showing input search field empty but shows filterd company name. To enhance the user experience and avoid any potential confusion, please consider either retaining the search term in the input field after a user has performed a search and navigated back to the home page from a company details page or clearing the search input and displaying all companies when a user returns to the home page from a details page. This adjustment will maintain consistency and clarity for users transitioning between the details page and the home page.

[Stock-App (1).webm](https://github.com/AliRezaBmeDu/metrics-webapp/assets/55840999/06bdb388-66a7-46ef-80e5-72d1bacc6d9c)


You can implement either of these options based on your design preferences. Both options help maintain consistency and clarity for users transitioning between the details page and the home page.

Reply:
done